pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -235,6 +236,25 @@ cpdef inline int64_t cast_from_unit(object ts, object unit) except? -1:
     return <int64_t> (base *m) + <int64_t> (frac *m)
 
 
+cpdef match_iso_format(object ts):
+    """
+    Match a provided string against an ISO 8601 pattern, providing a group for
+    each ``Timedelta`` component.
+    """
+    pater = re.compile(r"""P
```

Comment:
you prob want to compile this pattern in a module level variable to avoid repeatedly doing this.